pitop/pma/encoder_motor.py

```diff
@@ -56,6 +57,8 @@ def __init__(self, port_name, forward_direction, braking_type=BrakingType.COAST,
         self._prev_time_dist_cnt = time.time()
         self._previous_reading_odometer = 0
 
+        weakref.finalize(self._motor_core, self._motor_core.cleanup)
```

Comment:
This looks like an `atexit` style function - is it possible to change the syntax to do the same thing, and maintain consistency?

Reply:
I'm not sure, I got this suggestion from [here ](https://stackoverflow.com/a/42907819/12423177) - I'm not that familiar with this stuff but it seems from other sources too (like [this](https://stackoverflow.com/questions/51513386/how-to-ensure-the-del-function-is-called-on-a-python-class-as-is-commonly-b)) that I would need to use `with` if I wanted to use a __del__ method so if someone used the controller without that then they'd not be guaranteed a cleanup. Is that what you mean? Or can atexit be used as a drop in replacement for the weakref.finalize command I have here?

Reply:
[this answer](https://stackoverflow.com/a/51513387/12423177) using atexit was downvoted for some garbage collection stuff, maybe you'll understand it more